.cloudbuild/library_generation/library_generation.Dockerfile

```diff
@@ -11,107 +11,197 @@
+RUN cp ./owl-bot-bin /owl-bot-bin
+
+# Creates the generator jar
+# maven:3.8.6-openjdk-11-slim
+FROM us-docker.pkg.dev/artifact-foundry-prod/docker-3p-trusted/maven@sha256:2cb7c73ba2fd0f7ae64cfabd99180030ec85841a1197b4ae821d21836cb0aa3b  as ggj-build
 
 WORKDIR /sdk-platform-java
 COPY . .
 # {x-version-update-start:gapic-generator-java:current}
 ENV DOCKER_GAPIC_GENERATOR_VERSION="2.45.1-SNAPSHOT" 
 # {x-version-update-end}
 
-RUN mvn install -B -ntp -DskipTests -Dclirr.skip -Dcheckstyle.skip
-RUN cp "/root/.m2/repository/com/google/api/gapic-generator-java/${DOCKER_GAPIC_GENERATOR_VERSION}/gapic-generator-java-${DOCKER_GAPIC_GENERATOR_VERSION}.jar" \
-  "./gapic-generator-java.jar"
+# use Docker Buildkit caching for faster local builds
+RUN --mount=type=cache,target=/root/.m2 mvn install -B -ntp -T 1.5C \
+    -DskipTests -Dclirr.skip -Dcheckstyle.skip -Djacoco.skip -Dmaven.test.skip \
+    -Dmaven.site.skikip -Dmaven.javadoc.skip -pl gapic-generator-java -am
 
-# build from the root of this repo:
-FROM gcr.io/cloud-devrel-public-resources/python
+RUN --mount=type=cache,target=/root/.m2 cp "/root/.m2/repository/com/google/api/gapic-generator-java/${DOCKER_GAPIC_GENERATOR_VERSION}/gapic-generator-java-${DOCKER_GAPIC_GENERATOR_VERSION}.jar" \
+  "/gapic-generator-java.jar"
 
-SHELL [ "/bin/bash", "-c" ]
+# Builds the python scripts in library_generation
+# python:3.11-alpine
+FROM us-docker.pkg.dev/artifact-foundry-prod/docker-3p-trusted/python@sha256:0b5ed25d3cc27cd35c7b0352bac8ef2ebc8dd3da72a0c03caaf4eb15d9ec827a as python-scripts-build
 
+# This will use GOOGLE_APPLICATION_CREDENTIALS if passed in docker build command.
+# If not passed will leave it unset to support GCE Metadata in CI builds
+ARG GOOGLE_APPLICATION_CREDENTIALS
+
+RUN apk add bash curl
+
+# Install gcloud to obtain the credentials to use the Airlock repostiory
+RUN curl -sSL https://sdk.cloud.google.com | bash -e
+ENV PATH $PATH:/root/google-cloud-sdk/bin
+
+
+# Configure the Airlock pip package repository
+RUN pip install keyrings.google-artifactregistry-auth -i https://pypi.org/simple/
+COPY .cloudbuild/library_generation/image-configuration/airlock-pypirc /root/.pypirc
+COPY .cloudbuild/library_generation/image-configuration/airlock-pip.conf /etc/pip.conf
+RUN chmod 600 /root/.pypirc /etc/pip.conf
+
+COPY library_generation /src
+
+# install main scripts as a python package
+WORKDIR /src
+
+RUN --mount=type=secret,id=credentials python -m pip install --target /usr/local/lib/python3.11 -r requirements.txt
+RUN python -m pip install --target /usr/local/lib/python3.11 .
+
+# Final image. Installs the rest of the dependencies and gets the binaries
+# from the previous stages. We use the node base image for it to be compatible
+# with the standalone binary owl-bot compiled in the previous stage
+# node:22.1-alpine
+FROM us-docker.pkg.dev/artifact-foundry-prod/docker-3p-trusted/node@sha256:487dc5d5122d578e13f2231aa4ac0f63068becd921099c4c677c850df93bede8 as final
```

Comment:
Why can't we use a python base image?

Reply:
The standalone owlbot seems to need a few runtime libraries (`.so`) in the linux environment that are available in the node image by default (besides the node runtime). The python image doesn't have them. 
If we want to have a python image as base, I can try to use a python base and install node or whichever libraries the executable needs.

Reply:
From discussion with @blakeli0 and @JoeWang1127, let's use a vanilla Alpine based image.